src/CMakeLists.txt

```diff
@@ -39,6 +53,21 @@ set(libvalhalla_include_directories "" CACHE INTERNAL "")
 set(libvalhalla_link_objects "" CACHE INTERNAL  "")
 set(libvalhalla_link_libraries "" CACHE INTERNAL  "")
 
+# Setup warning flags. Note that these are only added to libvalhalla_compile_options if
+# ENABLE_COMPILER_WARNINGS=ON. Note that this needs to be done before the definition of
+# valhalla_module, otherwise LIBCXX_COMPILE_FLAGS will not available for downstream targets.
+
+if (ENABLE_COMPILER_WARNINGS)
+  message(STATUS "Enabling compiler warning flags")
+endif()
+
+function (cxx_add_warning_flags target)
+  target_add_compile_flags_if_supported(${target} PRIVATE -Wall -Wextra -Wconversion)
+  if (ENABLE_WERROR)
```

Comment:
Do we need this functionality (of enabling treating all warnings as error) as part of this function? The name `cxx_add_warning_flags` seems to indicate it will enable some (or all) warnings. But right now it internally also enables treating warning as error. Maybe have this as a separate function that user calls to enable this feature?  Eg: I might want to error out on a warning, but not necessarily enable ALL warnings.
Something like this is what I suggest:
```cmake
if (ENABLE_COMPILER_WARNINGS)
    enable_warning_flags ...
endif()
if (ENABLE_WERROR)
    enable_error_flags ...
endif()
```

Reply:
I think in this case, the addition of `-Werror` is very tightly coupled to the configuration of the warnings flags. I can't imagine someone wanting to set this flag separately outside of the scope of this function. 

Regarding the naming of the function: I'm being consistent with the naming scheme inherited from the cmake module.

Reply:
The help string for `ENABLE_WERROR` does not indicate that the user has to enable `ENABLE_COMPILER_WARNINGS`. Maybe have a note somewhere saying `ENABLE_WERROR` has no effect unless  `ENABLE_COMPILER_WARNINGS` is enabled. Rest of of it looks good 👍